File: patterns/looping-timers/rules-js/looping-timers.js
let ceilingTimer = null;

rules.JSRule({
  name: "Ceiling fan control",
  triggers: [triggers.ItemStateChangeTrigger("MotionSensor", undefined, "ON")],
  execute: (event) => {
    if (ceilingTimer !== null) return;

    const loop = () => {
      if (items.getItem("vTimeOfDay").state === "NIGHT") {
        let newState = "STAY";
        const current = parseFloat(items.getItem("CurrentTemp").state);
        const target = parseFloat(items.getItem("TargetTemp").state);
        if (current > target) newState = "ON";
        else if (current < target - 1) newState = "OFF";

        if (newState !== "STAY" && items.getItem("Fan").state !== newState) {
          items.getItem("Fan").sendCommand(newState);
        }

        ceilingTimer.reschedule(time.ZonedDateTime.now().plusSeconds(60));
      } else {
        ceilingTimer = null;
      }
    };

    ceilingTimer = actions.ScriptExecution.createTimer(time.ZonedDateTime.now(), loop);
  }
});

File: patterns/looping-timers/rules-python/looping_timers.py
import threading

from openhab import rule, Registry, logger
from openhab.triggers import ItemStateChangeTrigger

ceiling_timer = None


def loop():
    global ceiling_timer
    if str(Registry.getItem("vTimeOfDay").getState()) == "NIGHT":
        current_state = str(Registry.getItem("CurrentTemp").getState())
        target_state = str(Registry.getItem("TargetTemp").getState())

        if current_state not in ("NULL", "UNDEF") and target_state not in ("NULL", "UNDEF"):
            new_state = "STAY"
            current = float(current_state)
            target = float(target_state)
            if current > target:
                new_state = "ON"
            elif current < target - 1:
                new_state = "OFF"

            if new_state != "STAY" and str(Registry.getItem("Fan").getState()) != new_state:
                Registry.getItem("Fan").sendCommand(new_state)
                logger.info("Fan -> " + new_state)

        ceiling_timer = threading.Timer(60, loop)
        ceiling_timer.start()
    else:
        ceiling_timer = None


@rule(triggers=[ItemStateChangeTrigger("MotionSensor", state="ON")])
class CeilingFanControl:
    def execute(self, module, input):
        global ceiling_timer
        if ceiling_timer is not None:
            return
        self.logger.info("Schleife gestartet")
        loop()
